lab_2_keywords_cooccurrence/main.py

```diff
@@ -3,12 +3,46 @@
 Extract keywords based on co-occurrence frequency
 """
 from pathlib import Path
-from typing import Optional, Sequence, Mapping
+from typing import Optional, Sequence, Mapping, Any
+
 
 KeyPhrase = tuple[str, ...]
 KeyPhrases = Sequence[KeyPhrase]
 
 
+def correct_types(user: Any, expected_type: Any, can_be_empty: bool = False) -> bool:
+    """
+    Checks type of variable and compares it with expected type.
+    For dict and list checks whether their elements are empty (regulated with can_be_empty)
+    """
+    if not (isinstance(user, expected_type) and user):
+        return False
+    if expected_type == list:
+        for element in user:
+            if not element and can_be_empty is False:
+                return False
+    elif expected_type == dict:
+        for key, value in user.items():
+            if not (key and value):
+                return False
+    return True
+
+def check_dict(dictionary: Any, first_type: Any, second_type: Any, empty: bool) -> bool:
```

Comment:
why both have this function and also the possibility to check dictionaries in `correct_types`?